src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,93 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import duke.utils.Datetime;
+
+/**
+ * Represents Deadline objects.
+ * Inherits from the abstract <code>Task</code> class.
+ */
+public class Deadline extends Task {
+    public static final String DEADLINE_SYMBOL = "D";
+    public static final String DATE_FORMAT_INPUT = "dd-MM-yyyy HHmm";
+    public static final String DATE_FORMAT_OUTPUT = "MMM dd yyyy hh:mm a";
+    public static final String DEADLINE_BREAK = "/by";
+
+    private final Datetime datetime;
+
+    /**
+     * Constructor method.
+     *
+     * @param description the description of the <code>Deadline</code>.
+     * @param isCompleted the completion status of the <code>Deadline</code>.
+     * @param datetime the <code>LocalDateTime</code> the <code>Deadline</code> task is to be completed by.
+     */
+    public Deadline(String description, boolean isCompleted, LocalDateTime datetime) {
+        super(description, isCompleted);
+        this.datetime = new Datetime(datetime, DATE_FORMAT_INPUT, DATE_FORMAT_OUTPUT);
+    }
+
+    /**
+     * Creates a <code>Deadline</code> that is not completed.
+     *
+     * @param description the description of the <code>Deadline</code>.
+     * @param datetime the <code>LocalDateTime</code> the <code>Deadline</code> task is to be completed by.
+     * @return an uncompleted <code>Deadline</code>.
+     */
+    public static Deadline createDeadline(String description , LocalDateTime datetime) {
+        return new Deadline(description, false, datetime);
+    }
+
+    /**
+     * Converts the <code>Deadline</code> to a <code>String</code>.
+     *
+     * @return a <code>String</code> representing the <code>Deadline</code>.
+     */
+    @Override
+    public String toString() {
+        String stringDateTime = this.datetime.getOutputDatetimeString();
+        String byDatetime = String.format("(by: %s)", stringDateTime);
+        return "[" + DEADLINE_SYMBOL + "]" + toStringSuffix() + " " + byDatetime;
+    }
+
+    /**
+     * Checks if the other object is equivalent to <code>Deadline</code>.
+     *
+     * @param other the object to be compared to.
+     * @return <code>true</code> if both objects are equal.
+     */
```

Comment:
Would be good to define what qualifies as equal in the JavaDoc comment. i.e both name, time and date must be equal.